Makefile

```diff
@@ -14,33 +17,32 @@ setup-ci-image:
 	docker build -t acorn:v-ci .
 	docker save acorn:v-ci | docker exec -i $$(docker ps | grep k3s | awk '{print $$1}') ctr --address /run/k3s/containerd/containerd.sock images import -
 
-validate:
+GOLANGCI_LINT_VERSION := v1.51.1
+lint:
+	if ! command -v golangci-lint &> /dev/null; then \
+  		echo "Could not find golangci-lint, installing version $(GOLANGLINT_CI_VERSION)."; \
+		curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $$(go env GOPATH)/bin $(GOLANGCI_LINT_VERSION); \
+	fi
 	golangci-lint run
 
-validate-ci: setup-ci-env
-	go generate
-	go mod tidy
-	go run tools/gendocs/main.go
+# Run all validators
+validate: validate-code validate-docs
+
+# Runs linters and validates that all generated code is committed.
+validate-code: tools tidy generate lint gen-docs
 	if [ -n "$$(git status --porcelain)" ]; then \
 		git status --porcelain; \
 		echo "Encountered dirty repo!"; \
 		git diff; \
 		exit 1 \
-	;fi
+	;fi 
 
 test:
 	go test $(TEST_FLAGS) ./...
 
 goreleaser:
 	goreleaser build --snapshot --single-target --rm-dist
 
-setup-ci-env:
```

Comment:
I like the idea of keeping this and having `lint` call this. Reason being that we may need to add more tools like this in the future.

Thoughts?